src/toil/batchSystems/kubernetes.py

```diff
@@ -489,17 +489,59 @@ def _ourJobObjectList(self, onlySucceeded=False, limit=None):
         # https://github.com/kubernetes-client/python/issues/953. For now we
         # will just throw an error if we don't get to the end of the list in
         # time.
-        
-        if onlySucceeded: 
-            return self._try_kubernetes(self._api('batch').list_namespaced_job, self.namespace, 
-                                            label_selector="toil_run={}".format(self.runID), field_selector="status.successful==1").items
- 
-        return self._try_kubernetes(self._api('batch').list_namespaced_job, self.namespace, 
-                                            label_selector="toil_run={}".format(self.runID)).items
+
+        token = None
+
+        while True:
+            # We can't just pass e.g. a None continue token when there isn't
+            # one, because the Kubernetes module reads its kwargs dict and
+            # cares about presence/absence. So we build a dict to send.
+            kwargs = {}
+            
+            if token is not None:
+                kwargs['_continue'] = token
+
+            if onlySucceeded: 
+                results =  self._try_kubernetes(self._api('batch').list_namespaced_job, self.namespace, 
+                                                label_selector="toil_run={}".format(self.runID), field_selector="status.successful==1", **kwargs).items
+            else:
+                results = self._try_kubernetes(self._api('batch').list_namespaced_job, self.namespace, 
+                                                label_selector="toil_run={}".format(self.runID), **kwargs).items
+            yield results
```

Comment:
This is going to maker the iterator yield a series of lists, instead of all the items from the lists.

Reply:
You need [this loop](https://github.com/DataBiosphere/toil/pull/3233/files#diff-183d11f3af830162fadf362f8c641fdfL524-L527) back, except without the `_isJobOurs` check.